lib/private/utils.bzl

```
@@ -136,6 +136,42 @@ def _file_exists(path):
     file_glob = native.glob([file_rel], exclude_directories = 1, allow_empty = True)
     return len(file_glob) > 0
 
+def _default_size(size, timeout):
+    """Provide a sane default for *_test size attribute.
+
+    I've filed https://github.com/bazelbuild/bazel/issues/22355 to fix this upstream.
+
+    The [test-encyclopedia](https://bazel.build/reference/test-encyclopedia) says:
+
+    > Tests may return arbitrarily fast regardless of timeout.
+    > A test is not penalized for an overgenerous timeout, although a warning may be issued:
+    > you should generally set your timeout as tight as you can without incurring any flakiness.
+
+    However Bazel's default for timeout is medium, which is dumb given this guidance.
+
+    It also says:
+
+    > Tests which do not explicitly specify a timeout have one implied based on the test's size as follows
+
+    Therefore if size is specified, we should allow timeout to take its implied default.
+    If neither is set, then we can fix Bazel's wrong default here to avoid warnings under
+    `--test_verbose_timeout_warnings`.
+
+    This function can be used in a macro which wraps a testing rule.
+
+    Args:
+        size: the size attribute of a test target
+        timeout: the timeout attribute of a test target
+
+    Returns:
+        "short" if neither is set, otherwise timeout
```

Nit (`short` is a timeout, not a `size` IIUC):
```suggestion
        "small" if neither is set, otherwise timeout
```

---

I just realized a better way to do it, so I completely removed this method.